### PR TITLE
chore: add production release GitHub action

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

Major changes:
- Add release workflow to concentrate all actions within one workflow
- Create individual workflows to deploy web-app per environment and PRs

Minor changes:
- Upgrade GitHub actions versions to the latest is the _touched_ files
- Add ~5~ 15 seconds delay when creating the SSH tunnel

Green build => https://github.com/MobilityData/mobility-feed-api/actions/runs/7994550255

After merging we will need to test the release event.

GitHub workflow screenshot:
Screenshot 2024-02-21 at 2 55 23 PM

**Expected behavior:** 

Explain and/or show screenshots for how you expect the pull request to work in your testing (in case other devices exhibit different behavior).

**Testing tips:**

- Check build passed

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [ ] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: .github/workflows/api-deployer.yml [hunk 1]
```diff
@@ -130,6 +130,7 @@ jobs:
         with:
           name: terraform-artifact-plan.txt
           path: infra/artifact-registry/terraform-plan.txt
+          overwrite: true
```
Comment (davidgamez) [inline .github/workflows/api-deployer.yml:133]: This is needed when running multiple workflows under the same parent workflow.

**Comment by emmambd** [comment]
@davidgamez When does this get triggered? Per merge? Manual button?

**Comment by davidgamez** [comment]
> @davidgamez When does this get triggered? Per merge? Manual button?

When we create a release.

File: .github/workflows/api-prod.yml [hunk 2]
```diff
@@ -21,12 +21,3 @@ jobs:
       GCP_MOBILITY_FEEDS_SA_KEY: ${{ secrets.PROD_GCP_MOBILITY_FEEDS_SA_KEY }}
       OAUTH2_CLIENT_ID: ${{ secrets.PROD_MOBILITY_FEEDS_OAUTH2_CLIENT_ID}}
       OAUTH2_CLIENT_SECRET: ${{ secrets.PROD_MOBILITY_FEEDS_OAUTH2_CLIENT_SECRET}}
-
-  integration-tests:
```
[the remark this answers is not in the document]
Comment (jcpitre) [reply .github/workflows/api-prod.yml:25]: Because it takes too long?

File: .github/workflows/web-app-deployer.yml [hunk 3]
```diff
@@ -211,10 +209,10 @@ jobs:
 
       - name: Select Firebase Project
         working-directory: web-app
-        run: npx firebase use ${{  env.FIREBASE_PROJECT }}
+        run: npx firebase use ${{  inputs.FIREBASE_PROJECT }}
 
-      - name: Deploy to Firebase Hosting (${{ env.FIREBASE_PROJECT }})
-        if: ${{ env.FIREBASE_PROJECT != 'dev' }}
+      - name: Deploy to Firebase Hosting (${{ inputs.FIREBASE_PROJECT }})
+        if: ${{ github.event_name != 'pull_request' }}
```
Comment (jcpitre) [inline .github/workflows/web-app-deployer.yml:215]: This will always happen since you've removed the `pull_request` trigger, no?

**Review by jcpitre — CHANGES_REQUESTED** [review]
Use of `pull_request` in web-app-deployer.yml should be examined.

**Comment by davidgamez** [comment]
> Use of `pull_request` as an event in web-app-deployer.yml should be examined.

That's correct. I'll add GitHub action to make clear the behavior of the Web deployment in the PR.

File: .github/workflows/api-prod.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidgamez) [reply .github/workflows/api-prod.yml:25]: It's too long(as expected). Also, it helps to easily distinguish failures from _integration tests_ or _API deployment_.

File: .github/workflows/web-app-deployer.yml (hunk 3 above)
Comment (davidgamez) [reply .github/workflows/web-app-deployer.yml]: Fixed, added an additional GH action

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+jobs:
+  bd-deployment:
+    name: Database Deployment
+    uses: ./.github/workflows/db-prod.yml
+    secrets: inherit
+  bd-update:
+    name: Database Update
+    needs: bd-deployment
+    uses: ./.github/workflows/db-update-prod.yml
+    secrets: inherit
```
Comment (cka-y) [inline .github/workflows/release.yml:15]: 😍 didn't know we could do that

**Review by cka-y — APPROVED** [review]
🥳

**Review by jcpitre — APPROVED** [review]
LGTM!

**Pull request merged**